### PR TITLE
Improved maps

### PR DESCRIPTION
**Pull request by tigerw**
Tada!

### CODE REVIEW

File: src/BlockID.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ enum ENUM_BLOCK_ID  // tolua_export
 	E_BLOCK_TORCH = 50,
 	E_BLOCK_FIRE = 51,
 	E_BLOCK_MOB_SPAWNER = 52,
-	E_BLOCK_WOODEN_STAIRS = 53,
+	E_BLOCK_OAK_WOOD_STAIRS = 53,
```
Comment (NiLSPACE) [inline src/BlockID.h]: This could possibly break plugins.

File: src/BlockID.h [hunk 2]
```diff
@@ -120,7 +120,7 @@ enum ENUM_BLOCK_ID  // tolua_export
 	E_BLOCK_PUMPKIN_STEM = 104,
 	E_BLOCK_MELON_STEM = 105,
 	E_BLOCK_VINES = 106,
-	E_BLOCK_FENCE_GATE = 107,
+	E_BLOCK_OAK_FENCE_GATE = 107,
```
Comment (NiLSPACE) [inline src/BlockID.h]: Same here.

**Comment by NiLSPACE** [comment]
The item renaming can break plugins. Perhaps add the old names at the end of the list with a comment telling they are deprecated?

**Comment by tigerw** [comment]
Pfft, backwards compatibility is overrated.

Is there some way to tell plugins that they're using an outdated constant?

**Comment by SamJBarney** [comment]
No there isn't.

**Comment by NiLSPACE** [comment]
Yes there is, but you'll have to change the `__index` metamethod on the global environment (_G) ;)

**Comment by tigerw** [comment]
Gentlemen, behold! An expert at work.

**Comment by tigerw** [comment]
...how might I do this?

**Comment by NiLSPACE** [comment]
I doubt you'd want to do that. By doing that you'd make a plugin call your function every time you'd try to get a global variable.

**Comment by tigerw** [comment]
I added the aliases for the values I thought would be currently in use, so I omitted the (DOUBLE_)NEW_STONE_STONE_SLAB ones.

Mergeable?

**Comment by NiLSPACE** [comment]
None of the GetMapBaseColourID functions are overridden. Is that intended?

**Comment by tigerw** [comment]
Whoops, thanks.

Interesting: `override` isn't actually needed to override virtual functions but allows the compiler to verify that you've overridden correctly...

**Comment by SamJBarney** [comment]
@tigerw the number that you are returning when getting the ColourID, can you turn them into an enum so we can see which color is actually being returned?

**Comment by tigerw** [comment]
I thought about that, however, the problem is that, as can be seen from [here](http://minecraft.gamepedia.com/Map_item_format), the base colours are various shades of brown/grey/green/etc. that really have no good textual description. I would say that looking at the colours in the wiki is better than reading identifiers such as "light brown", "brown", and "dark brown".

**Comment by SamJBarney** [comment]
I see. NVM.

**Comment by tigerw** [comment]
Merge?

**Comment by SphinxC0re** [comment]
Looks good. Maxim + merge :D

**Comment by madmaxoft** [comment]
Why did you remove all the empty lines between functions? They make the code more readable by making the function headers stand out visually.

**Pull request merged**